Samples/Samples/ViewModel/CalendarEventAddViewModel.cs

```diff
@@ -322,22 +376,74 @@ public bool IsMonthDaySpecific
                     {
                         SelectedRecurrenceMonthDay = 0;
                         SelectedRecurrenceMonthWeek = IterationOffset.First;
-                        SelectedMonthWeekRecurrenceDay = DayOfTheWeek.Monday;
+                        SelectedMonthWeekRecurrenceDay = CalendarDayOfWeek.Monday;
                     }
                 }
             }
         }
 
-        public ObservableCollection<DayOfTheWeekSwitch> RecurrenceDays { get; set; } = new ObservableCollection<DayOfTheWeekSwitch>()
+        bool IsUpdatingCheckBoxGroup { get; set; } = false;
+
+        void SetCheckBoxes(int bitFlagValue)
         {
-            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Monday },
-            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Tuesday },
-            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Wednesday },
-            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Thursday },
-            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Friday },
-            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Saturday },
-            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Sunday }
-        };
+            try
+            {
+                IsUpdatingCheckBoxGroup = true;
+                var currentVal = bitFlagValue;
+                var maxValue = (int)CalendarDayOfWeek.Saturday;
+                for (var i = maxValue; i > 0; i /= 2)
+                {
+                    RecurrenceDays[(int)Math.Log(i, 2)].IsChecked = currentVal >= i;
```

Comment:
Hmmm not sure we should be relying on >= here for this.
Wouldn't it be better to cycle through the RecurrenceDays array and set value based on logical & with bitFlagValue
eg
RecurrenceDays[i].IsChecked=(RecurrenceDays[i].Day & bitFlagValue)>0

Reply:
umm, I can't use just 'i' as the array value because that's currently being used as a bit flag in a sense, i.e. this for loops 'i' goes 64 > 32 > 16> 8 > 4> 2 > 1, That's why I've got the Math.Log.

Reply:
right but you don't need to  - you can just do a foreach in the RecurrenceDays array can't you?

Reply:
Altered to use HasFlag as it's a much cleaner check